include/drivers/clock_control/nrf_clock_control.h

```diff
@@ -61,4 +59,46 @@
 void nrf5_power_usb_power_int_enable(bool enable);
 #endif
 
+/**
+ * @brief Initialize LFCLK RC calibration.
+ *
+ * @param hfclk_dev HFCLK device.
+ */
+void nrf_clock_control_calibration_init(struct device *hfclk_dev);
+
+/**
+ * @brief Calibration interrupts handler
+ *
+ * Must be called from clock interrupt context.
+ */
+void nrf_clock_control_calibration_isr(void);
+
+/**
+ * @brief Stop calibration.
+ *
+ * Function called when LFCLK RC clock is being started. It starts calibration
+ *
+ * @param dev LFCLK device.
+ *
+ * @retval true if clock can be started.
+ * @retval false if clock was not stopped due to ongoing calibration and don't
+ *	   need to be started again because it is still on.
```

Comment:
Currently this is not entirely true. The function returns `false` when `cal_state == CAL_ACTIVE_OFF` and this means that the calibration is ongoing but after it completes, the clock will be stopped.

Reply:
it's actually true (at least now). If before calibration is done, calibration state will be changed to `idle` then in calibration done event clock will not be stopped.

That was initial intention but due to bug in the function it was misbehaving.